### PR TITLE
Move HTML sanitization from client to server side

### LINKED ISSUE #381: [BUG] Move HTML sanitization from client to server side
**What is the bug?**
Currently for headers and footers we store the HTML as the index, and the sanitization occurs on the client side. Moving the sanitization to the server side is more secure, and also storing the raw markdown vs the converted HTML.

### PR DESCRIPTION
**Pull request by davidcui1225**
Signed-off-by: David Cui <davidcui@amazon.com>

### Description
* Move HTML sanitization from client to server side
* Keep dompurify package version up to date
* Store markdown header & footer data vs converted HTML

### Issues Resolved
#381 

### Check List
- [ ] New functionality includes testing.
  - [x] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [ ] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by abbashus — DISMISSED** [review]
Thanks for this important change! :raised_hands:

File: dashboards-reports/public/components/main/report_definition_details/report_definition_details.tsx [hunk 1]
```diff
@@ -426,12 +426,12 @@ export function ReportDefinitionDetails(props: { match?: any; setBreadcrumbs?: a
       reportHeader:
         reportParams.core_params.hasOwnProperty('header') &&
         reportParams.core_params.header != ''
-          ? converter.makeMarkdown(reportParams.core_params.header)
```
Comment (ps48) [inline dashboards-reports/public/components/main/report_definition_details/report_definition_details.tsx:429]: Let's remove the import if not used anymore.

Comment (ps48) [reply dashboards-reports/public/components/main/report_definition_details/report_definition_details.tsx:429]: same for `report_details.tsx`

**Review by ps48 — DISMISSED** [review]
Minor changes 👍🏽

File: dashboards-reports/public/components/main/report_definition_details/report_definition_details.tsx (hunk 1 above)
Comment (davidcui1225) [reply dashboards-reports/public/components/main/report_definition_details/report_definition_details.tsx:429]: Removed import from both files

**Review by kavithacm — APPROVED** [review]

**Pull request merged**